opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -1093,61 +1091,15 @@ def make_model_config_json(
                 "framework_type": "sentence_transformers",
+                "pooling_mode": pooling_mode,
                 "all_config": json.dumps(all_config),
             },
         }
 
-        if pooling_mode is not None:
-            model_config_content["model_config"]["pooling_mode"] = pooling_mode
-        else:
-            pooling_config_json_file_path = os.path.join(
-                folder_path, "1_Pooling", "config.json"
-            )
-            if os.path.exists(pooling_config_json_file_path):
-                try:
-                    with open(pooling_config_json_file_path) as f:
-                        if verbose:
-                            print(
-                                "reading pooling config file from: "
-                                + pooling_config_json_file_path
-                            )
-                        pooling_config_content = json.load(f)
-                        pooling_mode_mapping_dict = {
-                            "pooling_mode_cls_token": "CLS",
-                            "pooling_mode_mean_tokens": "MEAN",
-                            "pooling_mode_max_tokens": "MAX",
-                            "pooling_mode_mean_sqrt_len_tokens": "MEAN_SQRT_LEN",
-                        }
-                        for mapping_item in pooling_mode_mapping_dict:
-                            if (
-                                mapping_item in pooling_config_content.keys()
-                                and pooling_config_content[mapping_item]
-                            ):
-                                pooling_mode = pooling_mode_mapping_dict[mapping_item]
-                                model_config_content["model_config"][
-                                    "pooling_mode"
-                                ] = pooling_mode
-                                break
-                        else:
-                            print(
-                                'Cannot find "pooling_mode_[mode]_token(s)" with value true in config.json file at ',
-                                pooling_config_json_file_path,
-                                ". Please add in the pooling config file or input in the argument for pooling_mode.",
-                            )
-
-                except IOError:
-                    print(
-                        "Cannot open in config.json file at ",
-                        pooling_config_json_file_path,
-                        ". Please check the config.json ",
-                        "file in the path.",
-                    )
-
         if normalize_result is not None:
             model_config_content["model_config"]["normalize_result"] = normalize_result
         else:
-            normalize_result_json_file_path = os.path.join(folder_path, "2_Normalize")
-            if os.path.exists(normalize_result_json_file_path):
+            if len(model._modules) >= 3 and isinstance(model._modules["2"], Normalize):
```

Comment:
I think otherwise we can say `normalize_result` = False?

Reply:
Edited!